backend/src/lib.rs

```diff
@@ -95,7 +119,7 @@ pub trait BackendFactory<F: FieldElement> {
         &self,
         pil: &'a Analyzed<F>,
         fixed: &'a [(String, Vec<F>)],
-        output_dir: Option<&'a Path>,
```

Comment:
This made it hard to change the path in the `CompositeBackend`. I don't see a good reason to pass a reference with a lifetime here anyway.

Reply:
sgtm